unlambda: false positive on builtins

```go
package main

import "fmt"

func main() {
	f := func(s []int) int { return len(s) }
	fmt.Println(f([]int{1, 2, 3}))
}
```

`gocritic` reports:

```
./builtins.go:6:7: unlambda: replace `func(s []int) int { return len(s) }` with `len`
```

However making the suggested change produces code that does not compile:

```
./builtins.go:6:7: use of builtin len not in function call
```

--- a/checkers/builtinShadow_checker.go
+++ b/checkers/builtinShadow_checker.go
@@ -16,68 +16,17 @@ func init() {
 	info.After = `length := 10`
 
 	collection.AddChecker(&info, func(ctx *lintpack.CheckerContext) lintpack.FileWalker {
-		builtins := map[string]bool{
-			// Types
-			"bool":       true,
-			"byte":       true,
-			"complex64":  true,
-			"complex128": true,
-			"error":      true,
-			"float32":    true,
-			"float64":    true,
-			"int":        true,
-			"int8":       true,
-			"int16":      true,
-			"int32":      true,
-			"int64":      true,
-			"rune":       true,
-			"string":     true,
-			"uint":       true,
-			"uint8":      true,
-			"uint16":     true,
-			"uint32":     true,
-			"uint64":     true,
-			"uintptr":    true,
-
-			// Constants
-			"true":  true,
-			"false": true,
-			"iota":  true,
-
-			// Zero value
-			"nil": true,
-
-			// Functions
-			"append":  true,
-			"cap":     true,
-			"close":   true,
-			"complex": true,
-			"copy":    true,
-			"delete":  true,
-			"imag":    true,
-			"len":     true,
-			"make":    true,
-			"new":     true,
-			"panic":   true,
-			"print":   true,
-			"println": true,
-			"real":    true,
-			"recover": true,
-		}
-		c := &builtinShadowChecker{ctx: ctx, builtins: builtins}
-		return astwalk.WalkerForLocalDef(c, ctx.TypesInfo)
+		return astwalk.WalkerForLocalDef(&builtinShadowChecker{ctx: ctx}, ctx.TypesInfo)
 	})
 }
 
 type builtinShadowChecker struct {
 	astwalk.WalkHandler
 	ctx *lintpack.CheckerContext
-
-	builtins map[string]bool
 }
 
 func (c *builtinShadowChecker) VisitLocalDef(name astwalk.Name, _ ast.Expr) {
-	if _, isBuiltin := c.builtins[name.ID.String()]; isBuiltin {
+	if isBuiltin(name.ID.Name) {
 		c.warn(name.ID)
 	}
 }

--- a/checkers/testdata/unlambda/negative_tests.go
+++ b/checkers/testdata/unlambda/negative_tests.go
@@ -94,3 +94,13 @@ func complexCalls() {
 		return returnInt(x)
 	}
 }
+
+func builtins() {
+	_ = func(s []int) int { return len(s) }
+	_ = func(s []int) int { return cap(s) }
+}
+
+func typeConvert() {
+	_ = func(x int) int32 { return int32(x) }
+	_ = func(x int) int { return int(x) }
+}

--- a/checkers/unlambda_checker.go
+++ b/checkers/unlambda_checker.go
@@ -44,6 +44,9 @@ func (c *unlambdaChecker) VisitExpr(x ast.Expr) {
 	if callable == "" {
 		return // Skip tricky cases; only handle simple calls
 	}
+	if isBuiltin(callable) {
+		return // See #762
+	}
 	fnType := c.ctx.TypesInfo.TypeOf(fn)
 	resultType := c.ctx.TypesInfo.TypeOf(result.Fun)
 	if !types.Identical(fnType, resultType) {

--- a/checkers/utils.go
+++ b/checkers/utils.go
@@ -8,6 +8,60 @@ import (
 	"github.com/go-lintpack/lintpack"
 )
 
+var goBuiltins = map[string]bool{
+	// Types
+	"bool":       true,
+	"byte":       true,
+	"complex64":  true,
+	"complex128": true,
+	"error":      true,
+	"float32":    true,
+	"float64":    true,
+	"int":        true,
+	"int8":       true,
+	"int16":      true,
+	"int32":      true,
+	"int64":      true,
+	"rune":       true,
+	"string":     true,
+	"uint":       true,
+	"uint8":      true,
+	"uint16":     true,
+	"uint32":     true,
+	"uint64":     true,
+	"uintptr":    true,
+
+	// Constants
+	"true":  true,
+	"false": true,
+	"iota":  true,
+
+	// Zero value
+	"nil": true,
+
+	// Functions
+	"append":  true,
+	"cap":     true,
+	"close":   true,
+	"complex": true,
+	"copy":    true,
+	"delete":  true,
+	"imag":    true,
+	"len":     true,
+	"make":    true,
+	"new":     true,
+	"panic":   true,
+	"print":   true,
+	"println": true,
+	"real":    true,
+	"recover": true,
+}
+
+// isBuiltin reports whether sym belongs to a predefined identifier set.
+func isBuiltin(sym string) bool {
+	return goBuiltins[sym]
+}
+
 // isStdlibPkg reports whether pkg is a package from the Go standard library.
 func isStdlibPkg(pkg *types.Package) bool {
 	return pkg != nil && pkg.Path() == pkg.Name()

